src/components/Comparisons/Comparisons.jsx

```diff
@@ -65,57 +69,61 @@ function Comparisons (props) {
   // gets product info for each related product
   useEffect(() => {
     if (related) {
-      var finalArray = [];
-      related.forEach((product) => {
-        server.get('/product', {product_id: product})
-          .then((data) => {
-            finalArray.push(data.data);
-          })
-          .catch((err) => {
-            console.log(err);
-          })
-          .then(()=> {
-            if (finalArray.length === related.length) {
-              setProducts(finalArray);
-            }
-          })
-          .catch((err) => {
-            console.log(err);
-          })
-      })
+      if (related.length === 0) {
+        setProducts([]);
+      } else {
+        var finalArray = [];
+        related.forEach((product) => {
+          if (product !== props.prod.id) {
+            server.get('/product', {product_id: product})
+            .then((data) => {
+              finalArray.push(data.data);
+            })
+            .catch((err) => {
+              console.log(err);
+            })
+            .then(()=> {
+              if (finalArray.length === related.length) {
+                setProducts(finalArray);
+              }
+            })
+            .catch((err) => {
+              console.log(err);
+            })
+          }
+        })
+      }
     }
   }, [related])
 
   // checks if related item is out of stock. If it is, delete that item from related products
-  // works but if you click on too many items it'll max out the requests to the API... works for some products but not others...
-  // useEffect(() => {
-  //   if (products) {
-  //     related.forEach((item) => {
-  //       server.get('/products/styles', {product_id: item})
-  //       .then((data)=> {
-  //         if (data.data.results[0].skus.null) {
-  //           if (data.data.results[0].skus.null.quantity === null) {
-  //             console.log(`${item} is out of stock`);
-  //             var index = related.indexOf(item);
-  //             var copy = related.slice();
-  //             copy.splice(1, index);
-  //             setRelated(copy);
-  //           }
-  //           console.log('all related items in stock')
-  //         }
-  //       })
-  //       .catch((err) => {
-  //         console.log(err);
-  //       })
-  //     })
-  //   }
-  // }, [products])
-
-  //66643 seems to be out of stock... so may have to check if there are any in stock first because thumbnail and other properties are null
+  // works but if you click on too many items it'll max out the requests to the API... don't click on too many related products in a short time fram.
+  useEffect(() => {
+    if (products) {
+      related.forEach((item) => {
+        server.get('/products/styles', {product_id: item})
+        .then((data)=> {
+          if (data.data.results[0].skus.null) {
+            if (data.data.results[0].skus.null.quantity === null) {
+              console.log(`${item} is out of stock`);
```

Comment:
Not sure if this is needed? If we need to alert the user, a console log won't tell them - but probably just as effective if it doesn't show up? Not sure if the BRDs say something specific. Just a thought for reducing console.logs

Reply:
Done!